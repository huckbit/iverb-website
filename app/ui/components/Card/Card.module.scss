@import '../../../settings';

.cardWrapper {
  padding: 1.5rem 2rem;
  border-radius: 0.5rem;
  background-color: var(--secondary-bg);
  min-width: 18rem;
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.1);
  display: flex;
  align-items: center;
  justify-content: space-between;
  transition: all 0.3s ease-in-out;
  color: var(--primary-text);
  &:hover {
    cursor: pointer;
    transform: translateX(0.2rem);
    .chevronContainer {
      transform: translateX(0.5rem);
    }
  }

  @include breakpoint(large) {
    min-width: 14rem;
    min-height: 8.5rem;
  }
  @include breakpoint(colossal) {
    min-width: 25rem;
  }
}
.cardWrapperDark {
  @extend .cardWrapper;
  background-color: var(--secondary-bg);
}
.chevronContainer {
  transition: transform 0.3s ease-in-out;
}

.content {
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding: 1rem;
  margin-right: 4rem;
  flex-grow: 1;
}

.title {
  font-size: 1.5rem;
  font-weight: 700;
  color: var(--primary-text);
  font-family: playfair display, serif;
  font-style: italic;
}
