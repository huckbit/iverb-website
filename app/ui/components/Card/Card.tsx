import type { Verb } from '@lib/definitions';
import style from './Card.module.scss';
import { useTheme } from 'next-themes';
import { ChevronRightIcon } from '@heroicons/react/24/outline';
import { isLongString, truncateWithEllipsis } from '@global/utils';

export default function Card({ verb }: { verb: Verb }) {
  const { resolvedTheme } = useTheme();

  const { infinitive, presentParticiple, past, pastParticiple } = verb;
  return (
    <div className={resolvedTheme === 'light' ? style.cardWrapper : `${style.cardWrapperDark}`}>
      <div>
        <div>
          <h3 className={style.title}>{`To ${infinitive}`}</h3>
          <p>
            {isLongString(past) ? truncateWithEllipsis(past, 10) : past} - {isLongString(pastParticiple) ? truncateWithEllipsis(pastParticiple, 10) : pastParticiple}
          </p>
        </div>
      </div>
      <div className={style.chevronContainer}>
        <ChevronRightIcon className='w-6 h-6' />
      </div>
    </div>
  );
}
