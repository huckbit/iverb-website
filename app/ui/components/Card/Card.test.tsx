import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'next-themes';
import Card from './Card';
import mock from './Card.mock';

beforeAll(() => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
});

test('matches snapshot', () => {
  const container = render(<Card verb={mock} />);
  expect(container).toMatchSnapshot();
});

test('matches snapshot with dark theme', () => {
  const container = render(
    <ThemeProvider themes={['dark']}>
      <Card verb={mock} />
    </ThemeProvider>,
  );
  expect(container).toMatchSnapshot();
});

test('renders title correctly', () => {
  render(<Card verb={mock} />);
  expect(screen.getByRole('heading')).toBeInTheDocument();
});

test('renders verb details correctly', () => {
  render(<Card verb={mock} />);
  expect(screen.getByText('was - been')).toBeInTheDocument();
});

test('renders card with truncated past verb', () => {
  const verbWithLongPast = { ...mock, past: 'a very long verb name' };
  render(<Card verb={verbWithLongPast} />);
  expect(screen.getByText('a very lon... - been')).toBeInTheDocument();
});

test('renders card with truncated past verb', () => {
  const verbWithLongPast = { ...mock, pastParticiple: 'a very long verb name' };
  render(<Card verb={verbWithLongPast} />);
  expect(screen.getByText('was - a very lon...')).toBeInTheDocument();
});
