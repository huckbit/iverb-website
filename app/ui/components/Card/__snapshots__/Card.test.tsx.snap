// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardWrapperDark"
      >
        <div>
          <div>
            <h3
              class="title"
            >
              To be
            </h3>
            <p>
              was
               - 
              been
            </p>
          </div>
        </div>
        <div
          class="chevronContainer"
        >
          <svg
            aria-hidden="true"
            class="w-6 h-6"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m8.25 4.5 7.5 7.5-7.5 7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardWrapperDark"
    >
      <div>
        <div>
          <h3
            class="title"
          >
            To be
          </h3>
          <p>
            was
             - 
            been
          </p>
        </div>
      </div>
      <div
        class="chevronContainer"
      >
        <svg
          aria-hidden="true"
          class="w-6 h-6"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8.25 4.5 7.5 7.5-7.5 7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches snapshot with dark theme 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <script>
        !function(){try{var d=document.documentElement,n='data-theme',s='setAttribute';var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';d[s](n,'dark')}else{d.style.colorScheme = 'light';d[s](n,'light')}}else if(e){d[s](n,e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
      </script>
      <div
        class="cardWrapper"
      >
        <div>
          <div>
            <h3
              class="title"
            >
              To be
            </h3>
            <p>
              was
               - 
              been
            </p>
          </div>
        </div>
        <div
          class="chevronContainer"
        >
          <svg
            aria-hidden="true"
            class="w-6 h-6"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m8.25 4.5 7.5 7.5-7.5 7.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <script>
      !function(){try{var d=document.documentElement,n='data-theme',s='setAttribute';var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';d[s](n,'dark')}else{d.style.colorScheme = 'light';d[s](n,'light')}}else if(e){d[s](n,e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
    </script>
    <div
      class="cardWrapper"
    >
      <div>
        <div>
          <h3
            class="title"
          >
            To be
          </h3>
          <p>
            was
             - 
            been
          </p>
        </div>
      </div>
      <div
        class="chevronContainer"
      >
        <svg
          aria-hidden="true"
          class="w-6 h-6"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8.25 4.5 7.5 7.5-7.5 7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
